std/algorithm/searching.d

```diff
@@ -1792,27 +1799,44 @@ if (isRandomAccessRange!R1 && isBidirectionalRange!R2
     }
     // Stage 2: linear find
     size_t scout = needleLength - 1;
+outer:
     for (;;)
     {
         if (scout >= haystack.length)
-        {
-            return haystack[haystack.length .. haystack.length];
-        }
+            break;
         if (!binaryFun!pred(haystack[scout], needleBack))
         {
             ++scout;
             continue;
         }
         // Found a match with the last element in the needle
-        auto cand = haystack[scout + 1 - needleLength .. haystack.length];
-        if (startsWith!pred(cand, needle))
+        static if (isRandomAccessRange!R2)
         {
+            for (size_t j = scout + 1 - needleLength, k = 0; k < needleLength - 1; ++j, ++k)
```

Comment:
No effect on performance for ldc and dmd. [code](https://github.com/qznc/find-is-too-slow/commit/ed044054842522ccaa0d80006860dcf8ac6609d9)

I believe this is only worth it, if it leaves a register to other values. In this case, we probably need just as many registers (storing `h_base` instead of `j`) and we might have enough registers anyways, since there are not that many values necessary in this loop.